plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -204,25 +221,48 @@ func getTags(cert *x509.Certificate, location string) map[string]string {
 	return tags
 }
 
+// copied from refreshFilePaths() in plugins/inputs/file/file.go
+func (c *X509Cert) expandFilePathsToUrls() error {
```

Comment:
Can we change this function to return the overall list of URLs instead of passing it by the structure? While you are at it you can do the `append(c.urls, c.globFilePathsToUrls...)` here by initializing the overall list with `c.urls`...